client/src/components/TreeDropdownField/tests/TreeDropdownField-test.js

```diff
@@ -1,684 +1,413 @@
+      valueObjects: [
+        {
+          id: 67
+        },
+        {
+          id: 12
+        },
+      ],
+      urlTree: 'foo'
+    }
+  };
+  render(<TreeDropdownField {...props}/>);
+  expect(addSelectedValues).toBeCalledWith(props.id, [{ id: 67 }, { id: 12 }]);
+});
 
-  describe('handleBack()', () => {
-    let event = null;
-
-    beforeEach(() => {
-      event = document.createEvent('Event');
-      props.visible = [5, 9, 26];
-      field = ReactTestUtils.renderIntoDocument(
-        <TreeDropdownField {...props} />
-      );
-    });
-
-    it('should go up one level in the path', () => {
-      field.handleBack(event);
-
-      expect(props.actions.treeDropdownField.setVisible).toBeCalledWith(props.id, [5, 9]);
-    });
+test('TreeDropdownField multi-select should not call selectedValues without a valueObjects', () => {
+  const addSelectedValues = jest.fn();
+  let props = makeProps({
+    findTreePath: jest.fn((data, value) => (value ? [4, value] : null))
   });
+  props = {
+    ...props,
+    actions: {
+      ...props.actions,
+      treeDropdownField: {
+        ...props.actions.treeDropdownField,
+        addSelectedValues
+      }
+    },
+    value: 67,
+    data: {
+      multiple: true,
+      urlTree: 'foo'
+    }
+  };
+  render(<TreeDropdownField {...props}/>);
+  expect(addSelectedValues).not.toBeCalled();
+});
 
-  describe('handleNavigate()', () => {
-    let event = null;
-
-    beforeEach(() => {
-      event = document.createEvent('Event');
-      event.nativeEvent = { stopImmediatePropagation: jest.fn() };
-      props.visible = [5, 9];
-      field = ReactTestUtils.renderIntoDocument(
-        <TreeDropdownField {...props} />
-      );
-    });
-
-    it('should go to page twenty-six', () => {
-      field.handleNavigate(event, 26);
-
-      expect(props.actions.treeDropdownField.setVisible).toBeCalledWith(props.id, [5, 9, 26]);
-    });
-
-    it('should go to page twenty-seven', () => {
-      field.handleNavigate(event, 27);
-
-      expect(props.actions.treeDropdownField.setVisible).toBeCalledWith(props.id, [7, 27]);
-    });
+test('TreeDropdownField should clear search on reset', async () => {
+  let doResolve;
+  const promise = new Promise((resolve) => {
+    doResolve = resolve;
   });
+  const setSearch = jest.fn(() => doResolve());
+  const { container } = render(
+    <TreeDropdownField {...makeProps({
+      actions: {
+        ...makeProps().actions,
+        treeDropdownField: {
+          ...makeProps().actions.treeDropdownField,
+          setSearch
+        }
+      },
+    })}
+    />
+  );
+  const input = container.querySelector('.treedropdownfield__value-container input');
+  // arrow down will trigger search reset
+  fireEvent.keyDown(input, { key: 'ArrowDown', keyCode: 40 });
+  // ensure that 500 TreeDropdownField::SEARCH_DELAY is exceeded
+  act(() => jest.runAllTimers());
+  await promise;
+  expect(setSearch).toBeCalled();
+});
 
-  describe('getDropdownOptions()', () => {
```

Comment:
These were all testing the output of a class method that was passed as a prop to a react-select component